### ISSUE TITLE
Invis.vbs - Detected as threat and deleted by Antivirus softwares

### ISSUE THREAD

**Issue by jaganrv** [Digital-Sapphire/PyUpdater#208]
The invis.vbs file created to run the batch file in the background is detected by antivirus softwares as a threat and instantly deleted before execution resulting in following error.
![Untitled](https://user-images.githubusercontent.com/4150909/61518456-2f912a80-aa27-11e9-8d3e-02a0867c5ba5.png)

**Comment by JMSwag** [comment]
@jaganrv I'm thinking this will need a new look at the logic behind updating on windows. I'll look into this.

**Comment by mcsimps2** [comment]
I've also run into this problem on several computers who are running my program; the anti-virus programs don't seem to be particularly fond of the vbs script

**Comment by JMSwag** [comment]
@jaganrv @mcsimps2 Exclude the file from the scan.

**Comment by mcsimps2** [comment]
I've been using a custom update mechanism on Windows that works as follows to get around this antivirus restriction, since I cannot always add this as a whitelist rule to a computer's AV:

(1) Rename the currently running executable, `win.exe`, to `win.exe.old` and set the hidden file attribute on `win.exe.old`.  Note this is perfectly valid to do on a running Windows process - you can call `os.rename(sys.executable, sys.exectuable + ".old")` but not `os.move`.
(2) Download and extract the updated executable to `win.exe` since that file name is no longer taken
(3) Launch the new executable and exit process

The downside is now there can be two executables existing on the current system at any moment in time, unless there's some extra logic to go ahead and remove `win.exe.old`.  However, `win.exe.old` is hidden, so it doesn't appear to users.

I've been using this method with success in my own fork of PyUpdater.  My particular .exe file is running as a Windows service, where this method is also working well.

I know that not all users may like this method, but I would still be happy to add this to the codebase in case there are users that need to get around the vbs/batch file restriction, especially with antivirus detection.  Perhaps the ability to specify the restart method would be a working solution?

E.g.
```
extract_restart() // Default method pyupdater.BATCH, using the old update method
extract_restart(method=pyupdater.RENAME) // Uses the method described above
```

@JMSwag Would this be acceptable?  If so, I would be happy to add my alternative update code to the codebase and open up a PR.

**Comment by JMSwag** [comment]
@mcsimps2 Yes, and I like the minimal change to the API. The strategy has been used in [go-update](https://github.com/inconshreveable/go-update).

**Comment by JMSwag** [comment]
Closed in #227
